main.go

```diff
@@ -104,10 +106,16 @@ func main() {
 	adds := addCandidates.Difference(updates, Updatable)
 	deletes := deleteCandidates.Difference(updates, Updatable)
 
+	if len(deletes) > 0 && leaveUnknown {
+		fmt.Fprintf(stdout, "%d unknown records left untouched\n", len(deletes))
+		deletes = deletes[:0]
+	}
+
 	numChanges := len(updates) + len(adds) + len(deletes)
 
 	if numChanges > 0 && !yes {
 		if len(deletes) > 0 {
+			numChanges = numChanges - len(deletes)
```

Comment:
I don't think this line is needed after your newest changes? I believe it would make the number of changes negative if there was only records to delete.

Reply:
Nope. Not negative. Just zero ;)